### DISCUSSION TITLE
How to rotate a text before drawing

### DISCUSSION THREAD

**Discussion by ggolda** [SixLabors/ImageSharp.Drawing#146] [Q&A] [answered]
Is it possible to rotate a text that is being drawn atop of an image? Can't find a method that accepts an angle for rotation.

**Comment by tocsoft** [comment, marked as answer]
```csharp
image.Mutate(x => x
          .SetDrawingTransform(Matrix3x2Extensions.CreateRotationDegrees(45, new PointF(50, 50))) // set a transform to be applied to all drawing operations
          .DrawText("foo bar", font, Color.Black, new PointF(50, 50))); // draw the text
```

The one gotcha you might hit is the rotation origin of the matrix thus you will most likely need/want to set the point in the `CreateRotationDegrees(angle, point)` call to the same as the text drawing point to get the result you expect otherwise we will end up rotating around `0,0` (top left) and you can end up with text in odd places.

This API can be used to apply full matrix transforms to the items being drawn so can also be used for skewing, translating/offsetting, and scaling and as its a transformation matrix you can do them all at once.

or alternatively using `DrawingOptions` directly you do it thusly:
```csharp
  image.Mutate(x => x
            .DrawText(
                new DrawingOptions { 
                    Transform = Matrix3x2Extensions.CreateRotationDegrees(45, new PointF(50, 50))
                },  
               "foo bar", 
               font, 
               Color.Black, 
               new PointF(50, 50)));
```

**Reply by tocsoft** [reply]
@ggolda I moved this to the correct repo as it was not a Fonts question but a Drawing one.

**Reply by ggolda** [reply]
@tocsoft thanks for a great answer 👍

**Reply by vrmvgf** [reply]
I used CreateRotationDegrees(angle, point), but the letters didn't fully displayed, liked this:
![3MbChII](https://user-images.githubusercontent.com/95868784/145429731-a5566285-18c0-46f9-be9a-c0166fb6d3e5.jpeg)

Did I do something wrong?

```
           var g = new SixLabors.ImageSharp.Image<Rgba32>(1000, 1000);
            g.Mutate(x => x.Fill(Color.White));
            var fontpath = new FontCollection().Install(System.IO.Path.Combine(System.Environment.GetEnvironmentVariable("windir"), "Fonts/arialn.TTF"));
            var font = new SixLabors.Fonts.Font(fontpath, 90);
            g.Mutate(x => x.SetDrawingTransform(Matrix3x2Extensions.CreateRotationDegrees(270, new PointF(500.0F, 900.0F)))
            .DrawText("foo bar 123456", font, Color.Crimson, new PointF(500.0F, 900.0F)));
            g.Save("bar1.jpg");
```

**Reply by tocsoft** [reply]
nope, looks fine. Its going to be the same issue we have already reported as #175

**Reply by vrmvgf** [reply]
Thank you.
